classes/external/get_bigbluebuttons_by_courses.php

```diff
@@ -0,0 +1,150 @@
+<?php
+// This file is part of Moodle - http://moodle.org/
+//
+// Moodle is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// Moodle is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with Moodle.  If not, see <http://www.gnu.org/licenses/>.
+
+namespace mod_bigbluebuttonbn\external;
+
+use context_module;
+use external_api;
+use external_files;
+use external_format_value;
+use external_function_parameters;
+use external_multiple_structure;
+use external_single_structure;
+use external_util;
+use external_value;
+use external_warnings;
+
+defined('MOODLE_INTERNAL') || die();
+
+global $CFG;
+require_once($CFG->libdir . '/externallib.php');
+
+/**
+ * External service to get activity per course
+ *
+ * This is mainly used by the mobile application.
+ *
+ * @package   mod_bigbluebuttonbn
+ * @category  external
+ * @copyright 2018 onwards, Blindside Networks Inc
+ * @license   http://www.gnu.org/copyleft/gpl.html GNU GPL v3 or later
+ */
+class get_bigbluebuttons_by_courses extends external_api {
+    /**
+     * Describes the parameters for get_bigbluebuttonbns_by_courses.
+     *
+     * @return external_function_parameters
+     * @since Moodle 3.3
+     */
+    public static function execute_parameters() {
+        return new external_function_parameters (
+            array(
+                'courseids' => new external_multiple_structure(
+                    new external_value(PARAM_INT, 'Course id'), 'Array of course ids', VALUE_DEFAULT, array()
+                ),
+            )
+        );
+    }
+
+    /**
+     * Returns a list of bigbluebuttonbns in a provided list of courses.
+     * If no list is provided all bigbluebuttonbns that the user can view will be returned.
+     *
+     * @param array $courseids course ids
+     * @return array of warnings and bigbluebuttonbns
+     * @throws \coding_exception
+     * @throws \invalid_parameter_exception
+     * @since Moodle 3.3
```

Comment:
Is it?

Reply:
Not sure. I guess. This was in 2.4 already (in externallib.php). We can remove the comment.

Reply:
This is a new web service though so it isn't in 2.4.